tests/test_tasks/test_spatial.py

```diff
@@ -0,0 +1,117 @@
+                "slice_bbox": BBox(((730000, 4400000), (731000, 4401000)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 100, None), slice(0, 100, None)),
+            id="clean test",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (731000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (730990, 4400650)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 99, None), slice(0, 65, None)),
+            id="less regular test",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 200, None), slice(0, 100, None)),
+            id="random test",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "resolution": (10, 10),
+                "limit_x": (50, 100),
+                "limit_y": (100, 300),
+            },
+            (slice(50, 100, None), slice(100, 100, None)),
+            id="test limits",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(bbox=[500, 500, 600, 600], crs=CRS.WGS84),
+                "slice_bbox": BBox(bbox=[500, 500, 600, 600], crs=CRS.WGS84),
+                "resolution": (10, 10),
+            },
+            (slice(0, 10, None), slice(0, 10, None)),
+            id="test WGS84",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "size": (10, 10),
+            },
+            (slice(0, 10, None), slice(0, 10, None)),
+            id="test size",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "size": (10, 10),
+                "limit_x": (50, 100),
+                "limit_y": (100, 200),
+            },
+            (slice(50, 10, None), slice(100, 10, None)),
+            id="test size limit",
+        ),
+    ],
+)
+def test_get_array_slices(input_params: Dict[str, Any], expected: Tuple[slice, slice]) -> None:
+    assert get_array_slices(**input_params) == expected
+
+
+@pytest.mark.parametrize(
+    "input_params",
+    [
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
```

Comment:
this BBox repeats itself often, consider extracting it as a constant (at least for the `test_get_array_slices_invalid_input`)

Reply:
done